diff --git a/WinPort/src/sudo/sudo_client.cpp b/WinPort/src/sudo/sudo_client.cpp
--- a/WinPort/src/sudo/sudo_client.cpp
+++ b/WinPort/src/sudo/sudo_client.cpp
@@ -151,6 +151,7 @@ namespace Sudo
 				perror("chdir");
 			//if process doesn't have terminal then sudo caches password per parent pid
 			//so don't use intermediate shell for running it!
+			chdir(getenv("FARHOME"));
 			r = execlp("sudo", "-n", "-A", "-k", g_sudo_app.c_str(), ipc.c_str(), NULL);
 //			r = execlp(g_sudo_app.c_str(), g_sudo_app.c_str(), ipc.c_str(), NULL);
 			perror("execl");
